tests/ignite/engine/test_deterministic.py

```diff
@@ -21,6 +21,11 @@
 from tests.ignite.engine import BatchChecker, setup_sampler
 
 
+def test_dengine_setup_seed_div_by_zero():
+    with pytest.raises(ZeroDivisionError, match=r"integer division or modulo by zero"):
```

Comment:
Nice catch!
@vfdev-5 should we handle this `iter_counter=0` exception with our custom exception message?

Reply:
But `_setup_seed` is called internally with arguments which does not occur ZeroDivError

Reply:
oh, I wasn't clear, I meant catch it before `Zero..` with custom `ValueError`

```python
if iter_counter == 0:
    raise ValueError
```